clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -3506,6 +3538,68 @@ protected Set<String> maybeSendIngestionHeartbeat() {
     return failedPartitions;
   }
 
+  /**
+   *  Resubscribe operation by passing new version role and partition role to {@link AggKafkaConsumerService}. The action
+   *  for leader and follower replica will be handled differently.
+   */
+  private void resubscribe(
+      PartitionConsumptionState partitionConsumptionState,
+      TopicPartitionReplicaRole.VersionRole versionRoleFromConsumerAction) throws InterruptedException {
+    if (isLeader(partitionConsumptionState)) {
```

Comment:
We put leader flag inside TopicPartitionReplicaRole but sill use PCS to determine leader role? Which flag should be the source of truth when resubscription happens?

Reply:
For this, I assume these two should be consistent considering consumer actions will be handled sequentially, and it is good to leave some javadoc to make that clear.

Reply:
Here this class is not stateful but more like to bundle a lot of information to be passed to KCS.  We construct new `TopicPartitionReplicaRole` before subscribe in this resubscribe operation.